src/easylink/rule.py

```diff
@@ -1,15 +1,30 @@
+"""
+===============
+Snakemake Rules
+===============
+
+This module is responsible for generating the Snakemake rules to be run as well
+as writing them to the Snakefile.
+
+Note we have adopted the Snakemake term "rule" to refer to a singular component 
+in a Snakefile (i.e. in a Snakemake pipeline) that defines input files, output files,
+and the command to run to create those output files. These rules are generated
+dynamically as strings and appended to the Snakefile.
+"""
+
 import os
 from abc import ABC, abstractmethod
 from collections.abc import Callable
 from dataclasses import dataclass
 
 
 class Rule(ABC):
-    """Abstract class to define interface between Steps and Implementations.
+    """An abstract class to define the interface between :class:`Steps<easylink.step.Step>` and their :class:`Implementations<easylink.implementation.Implementation>`.
+
+    This class is responsible for converting the defined interfaces between ``Steps``
+    and ``Implementations`` to a Snakemake rule and writing it out to the Snakefile
+    to be run.
```

Comment:
I don't understand how this class is an interface between Steps and Implementations...

Reply:
I fought with that as well (I didn't modify this existing docstring much). I'll re-think this.

I guess really it acts more as the interface between the user's specification files + `Implementations` and the snakefile...